// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}

  rpc GetNextDay (NextDayRequest) returns (NextDayReply) {}

  rpc SumDaysDate (SumDaysRequest) returns (NextDayReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

// The request to get the next date
message NextDayRequest {
  int32 day = 1;
  int32 month = 2;
  int32 year = 3;
}

// The response containing the next date
message NextDayReply {
  int32 day = 1;
  int32 month = 2;
  int32 year = 3;
}

// The request to sum days to a date
message SumDaysRequest {
  int32 day = 1;
  int32 month = 2;
  int32 year = 3;
  int32 days = 4;
}
